protocols/relay/src/priv_client/handler.rs

```diff
@@ -499,27 +459,24 @@ impl ConnectionHandler for Handler {
                 );
                 match pending_request {
                     PendingRequest::Reserve { to_listener } => {
-                        self.active_reserve_requests.push_back(to_listener);
                         if self
                             .inflight_reserve_requests
-                            .try_push(outbound_hop::make_reservation(stream))
+                            .try_push(outbound_hop::make_reservation(stream), to_listener)
                             .is_err()
                         {
-                            tracing::warn!("Dropping outbound stream because we are at capacity")
+                            tracing::warn!("Dropping outbound stream because we are at capacity");
                         }
                     }
                     PendingRequest::Connect {
                         dst_peer_id,
                         to_dial: send_back,
                     } => {
-                        self.active_connect_requests.push_back(send_back);
-
                         if self
                             .inflight_outbound_connect_requests
-                            .try_push(outbound_hop::open_circuit(stream, dst_peer_id))
+                            .try_push(outbound_hop::open_circuit(stream, dst_peer_id), send_back)
```

Comment:
As an alternative to `FuturesTupleSet`, one could as well use `FuturesExt::map` to add the `send_back` to the return type of the `Future`.

I am fine either way.

Reply:
That is much simpler and we don't need a separate data structure! Great idea @mxinden !

Reply:
I just tried this and it doesn't work because the state is lost in case the future times out. We could solve that by removing the timeout bit from `futures-bounded` but then we'd have to repeat ourselves in saying `.timeout` on every future that we push into the set/map.